rasa_sdk/forms.py

```diff
@@ -848,9 +668,15 @@ async def validate(
         Returns:
             `SlotSet` events for every validated slot.
         """
+        slots_to_validate = await self.required_slots(
+            self.domain_slots(domain), dispatcher, tracker, domain
+        )
         slots: Dict[Text, Any] = tracker.slots_to_validate()
 
         for slot_name, slot_value in list(slots.items()):
+            if slot_name not in slots_to_validate:
+                continue
```

Comment:
It took me a while, but I think I understand why in `test_validation_action_outside_forms_with_form_active_loop`, the output events is not an empty list, because the slot persists in `slots` un-updated. Shouldn't the functionality be not to add an identical SlotSet to the tracker, but to return an empty list? So in this line, probably the slot needs to be removed from `slots` 🤔